Github runner: `ubuntu-latest (4.1.0)` is failing to install `odbc`

The github runner we use to test the minimum supported R version is failing to install `odbc`, causing a CI failure. The error is caused because a system dependency is missing: 

```
ℹ Building odbc 1.6.4.1
  ✖ Failed to build odbc 1.6.4.1 (380ms)
  Error: 
  ! error in pak subprocess
  Caused by error in `stop_task_build(state, worker)`:
  ! Failed to build source package odbc.
  Full installation output:
  * installing *source* package ‘odbc’ ...
  ** package ‘odbc’ successfully unpacked and MD5 sums checked
  staged installation is only possible with locking
  ** using non-staged installation
  Found pkg-config cflags and libs!
  R was installed withut a C++ compiler
  PKG_CFLAGS=
  PKG_LIBS=-lodbc 
  ./configure: 75: -E: not found
  ------------------------- ANTICONF ERROR ---------------------------
  Configuration failed because odbc was not found. Try installing:
   * deb: unixodbc-dev (Debian, Ubuntu, etc)
   * rpm: unixODBC-devel (Fedora, CentOS, RHEL)
   * csw: unixodbc_dev (Solaris)
   * pacman: unixodbc (Archlinux, Manjaro, etc)
   * brew: unixodbc (Mac OSX)
  To use a custom odbc set INCLUDE_DIR and LIB_DIR and PKG_LIBS manually via:
  R CMD INSTALL --configure-vars='INCLUDE_DIR=... LIB_DIR=... PKG_LIBS=...'
  --------------------------------------------------------------------
  ERROR: configuration failed for package ‘odbc’
  * removing ‘/tmp/RtmpqJ6Q0L/pkg-lib14625f5a7152/odbc’
  ---
  Backtrace:
  1. pak::lockfile_install(".github/pkg.lock")
  2. pak:::remote(function(...) { …
  3. err$throw(res$error)
  ---
  Subprocess backtrace:
   1. base::withCallingHandlers(cli_message = function(msg) { …
   2. get("lockfile_install_internal", asNamespace("pak"))(...)
   3. plan$install()
   4. pkgdepends::install_package_plan(plan, lib = private$library, num_workers = nw, …
   5. base::withCallingHandlers({ …
   6. pkgdepends:::handle_events(state, events)
   7. pkgdepends:::handle_event(state, i)
   8. pkgdepends:::stop_task(state, worker)
   9. pkgdepends:::stop_task_build(state, worker)
  10. base::throw(pkg_error("Failed to build source package {.pkg {pkg}}.", …
  11. | base::signalCondition(cond)
  12. global (function (e) …
  Execution halted
  Error: Process completed with exit code 1.

```

I've tried setting up a workflow step to install the dependency, this seems to succeed, but odbc still fails to install. When I switched to `mac-os` as a runner, odbc installed, but classInt failed to build from source. See #421 for my comments.

--- a/.github/workflows/R-CMD-check.yaml
+++ b/.github/workflows/R-CMD-check.yaml
@@ -25,7 +25,7 @@ jobs:
           - {os: ubuntu-latest,   r: 'release'}
           - {os: ubuntu-latest,   r: 'oldrel-1'}
           # And minimum supported version in DESCRIPTION
-          - {os: ubuntu-latest,   r: '4.1.0'}
+          - {os: windows-latest,   r: '4.1.0'}
 
     env:
       GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}

